### ISSUE TITLE
Optimize the training loop

### ISSUE THREAD

**Issue by staalexa** [mohanadkandil/rl-tum#12]
- [ ] Add epsilon-greedy policy for exploration.
- [ ] Fine-tune hyperparameters (learning rate, discount factor).
